### PR TITLE
11 get articles topic query

### PR DESCRIPTION
**Pull request by Sir1ys**

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -129,6 +129,51 @@ describe("/api/articles", () => {
...
   });
+
+  test("GET: 200 sends an array of articles to the client filtered by topic = mitch", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+
+        // checking the array
```
Comment (RichardNeat) [inline __tests__/index.test.js]: Please remove comments before committing

File: __tests__/index.test.js [hunk 2]
```diff
@@ -129,6 +129,51 @@ describe("/api/articles", () => {
...
+        // checking each article
+        expect(articles.every((article) => article.topic === "cats")).toBe(
+          true
+        );
+      });
+  });
+
+  //ERROR TESTING
+
+  test("GET: 400 when passed topic is invalid", () => {
```
Comment (RichardNeat) [inline __tests__/index.test.js]: Technically this isn't an invalid query, it's just that this query isn't found.

File: __tests__/index.test.js [hunk 3]
```diff
@@ -129,6 +129,51 @@ describe("/api/articles", () => {
...
+        // checking the array
+        expect(articles.length).toBe(1);
+
+        // checking each article
+        expect(articles.every((article) => article.topic === "cats")).toBe(
+          true
+        );
+      });
+  });
+
```
Comment (RichardNeat) [inline __tests__/index.test.js:271]: We don't really need a second happy path test for the same behaviour here, what would be useful is a test for valid and existent topic but happen to have no articles associated with it.

**Review by RichardNeat — CHANGES_REQUESTED** [review]
Just a couple of requested changes below please. Could you resubmit once done :)

File: models/articles-model.js [hunk 4]
```diff
@@ -13,14 +13,28 @@ exports.fetchArticle = (article_id) => {
   });
 };
 
-exports.fetchArticles = () => {
-  const query = `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url,
+exports.fetchArticles = (topic) => {
+  const topicGreenList = ["mitch", "cats", "paper"];
```
Comment (NRMSMITH) [inline models/articles-model.js]: Having the topics hardcoded means you won't be able to add any more later on if need be. How could you dynamically make sure that the topics are up to date with the database without having to edit this array each time?

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Have a look at my comment about keeping the topics up to date :)

File: __tests__/index.test.js [hunk 5]
```diff
@@ -124,11 +119,45 @@ describe("/api/articles", () => {
...
+  test("GET: 200 sends an empty array of articles to the client filtered by topic = paper, because there are not articles associated with this topic", () => {
+    return request(app)
+      .get("/api/articles?topic=paper")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+
```
Comment (pippaamy) [inline __tests__/index.test.js:267]: great testing

**Review by pippaamy — APPROVED** [review]
great job with those changes, go ahead and merge!

**Pull request merged**